sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -400,6 +400,13 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] with PredicateHelper {
     }
   }
 
+  private def findGroupColumn(alias: Alias): Option[AttributeReference] = alias match {
+    case alias @ Alias(attr: AttributeReference, name) if attr.name.startsWith("group_col_") =>
+      Some(AttributeReference(name, attr.dataType)(alias.exprId))
+    case Alias(alias: Alias, _) => findGroupColumn(alias)
```

Comment:
I feel it's a bit hacky to assume the Alias contains the actual grouping columns. How about we generate the name mapping (grouping attribute to actual group column name) during agg pushdown, put the name mapping in `ScanBuilderHolder`, and use the mapping to rewrite order by expression during limit pushdown?

Reply:
OK